src/plugins/discover/server/sample_data/register_sample_data.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { SampleDataRegistrySetup } from '@kbn/home-plugin/server';
+import { APP_ICON } from '../../common';
+import { getSavedSearchFullPathUrl } from '../../common/services/saved_searches';
+
+export function registerSampleData(sampleDataRegistry: SampleDataRegistrySetup) {
+  const linkLabel = i18n.translate('discover.sampleData.viewLinkLabel', {
+    defaultMessage: 'Discover',
+  });
+  const { addAppLinksToSampleDataset, getSampleDatasets } = sampleDataRegistry;
+  const sampleDatasets = getSampleDatasets();
+
+  sampleDatasets.forEach((sampleDataset) => {
+    const sampleSavedSearchObject = sampleDataset.savedObjects.find(
+      (object) => object.type === 'search'
+    );
+
+    if (sampleSavedSearchObject) {
+      addAppLinksToSampleDataset(sampleDataset.id, [
+        {
+          sampleObject: sampleSavedSearchObject,
+          getPath: getSavedSearchFullPathUrl,
```

Comment:
We might also want to define a time range which is greater than the default one (15 min). To do so we would need to support storing of time in saved search objects https://github.com/elastic/kibana/issues/9761 or to extend this URL with time range params.